orangecontrib/text/widgets/owwordcloud.py

```diff
@@ -421,16 +432,18 @@ def on_corpus_change(self, data):
         self.Info.clear()
 
         self.corpus_counter = Counter()
+        self.document_counter = Counter()
```

Comment:
If we remove the call to  `create_weight_list` from `on_corpus_change` (since it doesn't do anything connected to counts -- they are not computed yet), I think `document_counter` doesn't need to be a class variable, and results can be passed from `on_done` directly to `create_weight_list` function.